src/textUtils.c

```diff
@@ -59,39 +59,70 @@ size_t str_formatAdaAmount(uint64_t amount, char* out, size_t outSize)
 	return rawSize + suffixLength;
 }
 
-size_t str_formatUint64(uint64_t number, char* out, size_t outSize)
+static size_t stringifyUint64ToBufferReverse(uint64_t number, char* ptr, char* end)
 {
-	ASSERT(outSize < BUFFER_SIZE_PARANOIA);
-
-	char scratchBuffer[30];
-	char* ptr = BEGIN(scratchBuffer);
-	char* end = END(scratchBuffer);
-
 	// We print in reverse
 	// We want at least one iteration
-	int place = 0;
+	size_t printedChars = 0;
 	do {
 		WRITE_CHAR(ptr, end, '0' + (number % 10));
 		number /= 10;
-		place++;
+		++printedChars;
 	} while (number > 0);
+	return printedChars;
+}
 
-	// Size without terminating character
-	STATIC_ASSERT(sizeof(ptr - scratchBuffer) == sizeof(size_t), "bad size_t size");
-	size_t rawSize = (size_t) (ptr - scratchBuffer);
-
-	if (rawSize + 1 > outSize) {
+static void printReversedStringToBuffer(const char* reverted, size_t revertedSize, char* out, size_t outSize)
+{
```

Comment:
Shouldn't we assert that `strlen(reverted) == revertedSize`? Or remove the revertedSize argument altogether?

Reply:
IF it's closed by a 0 then sure. Previously it was not, because why would it be, it's not a string to display, just a temporary working area.